src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ListingGenerationController.java

```diff
@@ -321,8 +321,23 @@ public ResponseEntity<ListingCallbackResponse> generateHearingDocument(
 
         var listingData = listingRequest.getCaseDetails().getCaseData();
         var caseTypeId = listingRequest.getCaseDetails().getCaseTypeId();
-
-        return getResponseEntity(listingData, caseTypeId, userToken);
+        List<String> errorsList = new ArrayList<>();
+        boolean invalidCharsExist = listingService.checkInvalidCharsForAllParties(
+                listingRequest.getCaseDetails(), userToken, errorsList);
+        if (hasListings(listingData) && !invalidCharsExist) {
```

Comment:
From what I understand, here you have replaced
(hasListings(listingData) || (isAllowedReportType(listingData)
                    && (hasServedClaims(listingData) || hasSummaryAndDetails(listingData)))) compound condition with
**(hasListings(listingData) && !invalidCharsExist)**.  Have tested for the other conditions that you removed and are you sure nothing has broken? Refactoring the old set of conditions is fine. I am just trying to make sure nothing has broken.

Reply:
Other conditions are for reports only, hasServedClaims, hasSummaryAndDetails are checking fields for reports only. They should not be relevant for this event.